### PR TITLE
feat(PHPStan) add static analysis PHPStan.

### PR DESCRIPTION
**Pull request by Yozhef**

### CODE REVIEW

File: .github/workflows/static-analysis.yaml [hunk 1]
```diff
@@ -21,3 +21,19 @@ jobs:
...
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+
+            - name: Install dependencies
+              run: composer update --no-progress --no-interaction --prefer-dist
```
Comment (khaperets) [inline .github/workflows/static-analysis.yaml]: `composer install` instead `composer update`

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,6 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Call to an undefined method Symfony\\\\Component\\\\Config\\\\Definition\\\\Builder\\\\NodeDefinition\\:\\:children\\(\\)\\.$#"
```
Comment (khaperets) [inline phpstan-baseline.neon:4]: Add check `instanseof NodeDefinition` or ignore this line with `// @phpstan-ignore-line`

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,9 @@
+includes:
+    - phpstan-baseline.neon
+
+parameters:
+    level: 5
```
Comment (khaperets) [inline phpstan.neon]: level: 8

File: composer.json [hunk 4]
```diff
@@ -29,7 +29,9 @@
         "phpunit/phpunit": "^8.5.13||^9.4",
         "symfony/phpunit-bridge": "^5.2",
         "phpunit/php-code-coverage": "^7.0",
-        "squizlabs/php_codesniffer": "^3.5"
+        "squizlabs/php_codesniffer": "^3.5",
+        "phpstan/phpstan": "^0.12",
+        "phpstan/phpstan-phpunit": "^0.12"
```
Comment (khaperets) [inline composer.json:33]: What is this package for?

File: phpstan-baseline.neon [hunk 5]
```diff
@@ -0,0 +1,6 @@
+parameters:
+	ignoreErrors:
```
Comment (khaperets) [inline phpstan-baseline.neon:2]: There should be no ignoring errors, you will need to delete this file

File: composer.json (hunk 4 above)
Comment (Yozhef) [reply composer.json:33]: This need to check PHPUnit code
Example: 
```
It also contains this strict framework-specific rules (can be enabled separately):

Check that you are not using assertSame() with true as expected value. assertTrue() should be used instead.
Check that you are not using assertSame() with false as expected value. assertFalse() should be used instead.
Check that you are not using assertSame() with null as expected value. assertNull() should be used instead.
Check that you are not using assertSame() with count($variable) as second parameter. assertCount($variable) should be used instead.
```

I find it important to write tests correctly using the appropriate features.

File: phpstan.neon (hunk 3 above)
Comment (Yozhef) [reply phpstan.neon]: In my opinion, it is important to take small clear steps - just now the 8th level needs additional changes so set the minimum pass - and we will make improvements gradually.

File: phpstan-baseline.neon (hunk 5 above)
Comment (Yozhef) [reply phpstan-baseline.neon:2]: I consider it normal to ignore problems that are connected with the third-party client for example not the type hint class of the builder
 ```
 Method SymfonyBundles\RedisBundle\Tests\DependencyInjection\ConfigurationTest::testConfiguration() has no return typehint specified.
 ```

File: phpstan-baseline.neon (hunk 2 above)
Comment (Yozhef) [reply phpstan-baseline.neon:4]: I think it's not the best idea to ignore straight the whole builder is better to add to ignore 1 error with count === 1

File: phpstan.neon (hunk 3 above)
Comment (Yozhef) [reply phpstan.neon]: add max level.

**Review by khaperets — APPROVED** [review]

**Pull request merged**